File: src/TaskOTime/TaskOTime.TimeTrackingServices/IEventSource.vb
Namespace ActiveDevelop.TimeTrackingServices
    Public Interface IEventSource(Of EventSourceIDType As {Structure, IComparable(Of EventSourceIDType)})
        Inherits IComparable(Of IEventSource(Of EventSourceIDType))

        Property Id As EventSourceIDType
    End Interface
End Namespace

File: src/TaskOTime/TaskOTime.TimeTrackingServices/IEventSource.cs
using System;

namespace ActiveDevelop.TimeTrackingServices
{
    public interface IEventSource<EventSourceIDType> : IComparable<IEventSource<EventSourceIDType>> where EventSourceIDType : struct, IComparable<EventSourceIDType>
    {

        EventSourceIDType Id { get; set; }
    }
}